### PR TITLE
#159082373 User Receives descriptive signup/login messages.

### PR DESCRIPTION
**Pull request by abulojoshua1**
### What does this PR do?
- Provides the user with descriptive login/signup messages

### Description of Task to be completed?
- A user receives descriptive signup/login messages that help him/her understand what is happening during signup or login.

### How should this be manually tested?
- Test the login and signup routes on postman using this pull requests's hostname as `https://authorshaven-pr-14.herokuapp.com/`.

### Any background context you want to provide?
- I overrode and made changes to serializer.py class methods.
- Validated password for strength.
- Validated username by its length.
- Wrote test cases for validations that were not present.
- Made changes to the outputs of some test cases.

### What are the relevant pivotal tracker stories?
- [#159082373](https://www.pivotaltracker.com/story/show/159082373)

### CODE REVIEW

File: Procfile [hunk 1]
```diff
@@ -1 +1,2 @@
+release: python manage.py migrate --settings=authors.settings.staging
```
Comment (geofrocker) [inline Procfile:1]: What does this do?

Comment (abulojoshua1) [reply Procfile:1]: @geofrocker, I realized that the code in the staging environment is sometimes lagging behind on database migrations. So I decided to run make migrations each time the application is hosted so the missing database structure changes are made.

File: authors/apps/authentication/serializers.py [hunk 2]
```diff
@@ -23,8 +25,56 @@ class Meta:
...
+
+    # https://stackoverflow.com/questions/29813463/django-rest-framework-email-validation
+    def validate_username(self, username_var):
+        # check if user name already exists in the database
+        # check if username is between 6 to 255 chracters
+        # make sure username can only contain underscore characters included with alphanumeric characters
+        db_username = User.objects.filter(username=username_var)
+        if len(username_var) >= 25:
+            raise serializers.ValidationError(
+                "The username cannot be greater than twenty five characters.")
```
Comment (geofrocker) [inline authors/apps/authentication/serializers.py]: Combine the validation for minimum and maximum chars into one

File: authors/apps/authentication/serializers.py [hunk 3]
```diff
@@ -23,8 +25,56 @@ class Meta:
...
+        elif len(username_var) < 5:
+            raise serializers.ValidationError(
+                "The username cannot be less than six characters.")
+        elif not username_var.replace("_", "").isalpha():
```
Comment (geofrocker) [inline authors/apps/authentication/serializers.py]: use a regular expression here

File: authors/apps/authentication/serializers.py (hunk 2 above)
Comment (abulojoshua1) [reply authors/apps/authentication/serializers.py]: Ok @geofrocker, I'll do just that.

File: authors/apps/authentication/serializers.py (hunk 3 above)
Comment (abulojoshua1) [reply authors/apps/authentication/serializers.py]: Ok @geofrocker, I'll do just that.

File: Procfile (hunk 1 above)
Comment (geofrocker) [reply Procfile:1]: Okay

**Pull request merged**